internal/hooks/hook_iterator.go

```diff
@@ -0,0 +1,52 @@
+package hooks
+
+import (
+	"github.com/launchdarkly/go-server-sdk/v7/ldhooks"
+)
+
+type hookIterator struct {
```

Comment:
Hook iterator provides a forward/reverse iterator for iterating a collection of hooks.